libs/auth/src/java_card.ts

```diff
@@ -393,15 +420,17 @@ export class JavaCard implements Card {
     );
 
     /**
-     * If the user doesn't have a PIN:
+     * If the card doesn't have a PIN:
      * Verify that the card has a private key that corresponds to the public key in the card
      * VxAdmin cert
      *
-     * If the user does have a PIN:
+     * If the card does have a PIN:
      * Perform this verification later in checkPin because operations with this private key are
      * PIN-gated
      */
-    if (cardDetails.user.role === 'poll_worker') {
+    const cardDoesNotHavePin =
+      arePollWorkerCardDetails(cardDetails) && !cardDetails.hasPin;
+    if (cardDoesNotHavePin) {
```

Comment:
@benadida I imagine that this is a key point you'll be looking at in your security review - making sure that we can't somehow trick the system into skipping this card VxAdmin cert validation.

What makes this safe is the inconsistency handling that I describe in the PR description:

> Also of note, we're handing inconsistencies as follows:
> - If the system settings says that poll worker card PINs are enabled, and a previously programmed poll worker card without a PIN (but valid in all other respects) is inserted, it'll be treated as unprogrammed.
> - If the system settings says that poll worker card PINs are not enabled, and a previously programmed poll worker card with a PIN is inserted, it'll also be treated as unprogrammed.

So for example, if poll worker card PINs are not enabled, and we accordingly never call `checkPin` for them, could someone use a poll worker card with a PIN to completely skip validation of the card VxAdmin cert? The answer is no, because the card will be treated as unprogrammed here: https://github.com/votingworks/vxsuite/blob/b01a1687edd10e9ee2a13faa628c0a05597e8396/libs/auth/src/inserted_smart_card_auth.ts#L469-L478